### PR TITLE
Azure IoT hub MQTT client support

### PR DESCRIPTION
**Pull request by dgarske**
Uses wolfMQTT demo IoT hub with "demoDevice" device / key. Demonstrates how to create a SasToken using the key, uri and expiration. Demonstrates how to publish events and listening to devicebound messages.

### CODE REVIEW

**Comment by JacobBarthelmeh** [comment]
When I run ./configure && make I get some errors that pop up from examples/azure/azureiothub.c. Am seeing it with using GCC and CLANG on OSX.

**Comment by dgarske** [comment]
Yeah that's expected. You need be have Base64 encode enabled. I have notes in the README.md and at the top of the azureiothub.c. If you run the latest master and do a ./configure --enable-base64encode && sudo make install. Then you won't get any build errors. I suppose I can do a better job detecting the older wolfSSL and throw a nice warning and disable the Azure example. Let me clean that up a bit more.

**Comment by dgarske** [comment]
@JacobBarthelmeh fixed the build error with older wolfSSL. Thanks for the excellent review as usual.

File: examples/azure/azureiothub.c [hunk 1]
```diff
@@ -0,0 +1,654 @@
...
+        if (signal(SIGINT, sig_handler) == SIG_ERR) {
+            PRINTF("Can't catch SIGINT");
+        }
+#endif
+
+    #ifdef ENABLE_AZUREIOTHUB_EXAMPLE
+        azureiothub_test(&args);
+    #else
+        /* This example requires wolfSSL 3.9.1 or later with base64encode enabled */
+        PRINTF("Example not compiled in!");
```
Comment (JacobBarthelmeh) [inline examples/azure/azureiothub.c]: args return value should be set in the else case

Comment (dgarske) [inline examples/azure/azureiothub.c]: Thanks @JacobBarthelmeh. Just pushed a fix for this here and in the firmware examples to return EXIT_FAILURE in this case. Good catch!

**Comment by JacobBarthelmeh** [comment]
Thanks for the changes, looks good!

**Pull request merged**